app/controllers/sequences_controller.rb

```diff
@@ -44,6 +44,38 @@ def show
       @lineages = sequence.lineages(equate_il, true).to_a
     end
 
+    # create list for all ec_cross_reference numbers
+    ecnumbers = []
+    # get all ec_cross_references
+    ec_cross_numbers = @entries.map(&:ec_cross_references)
+    # get all ec_cross_reference numbers and puts them in array
+    ec_cross_numbers.each do |ecs|
+      ecs.each do |ec|
+        ecnumbers << ec.ec_number
+      end
+    end
+    # make list unique
+    uniqecnumbers = ecnumbers.uniq
+    # get all items from table EcNumber
+    ecdb = EcNumber.all
+    # add column names list
+    @column_name = ["name", "digit 1", "digit 2", "digit 3", "digit 4", "count"]
+    # nested list in which each row is stored
+    tp_ec_lca_table = []
+    # create a hash that contails all required data for visualization
+    @ec_lca_table = {}
+    # for each piece of an ec number get its enzymatical function
+    uniqecnumbers.each do |ecn|
+      tpe = ecn.split(".")
+      @ec_lca_table[ecn] = [ecdb.find_by(ec_number: tpe[0]+".-.-.-"),
```

Comment:
I think this will generate a huge number of queries. Better to first create a list of all ec numbers we want info on and then retrieve everything in a single query.


_[Original comment](https://github.ugent.be/unipept/unipept/pull/528#discussion_r3383) by @bmesuere on Wed Oct 21 2015 at 16:46._